include/sys/socket.h

```diff
@@ -387,6 +387,37 @@ int socket(int domain, int type, int protocol);
 */
 int getsockname(int socket, struct sockaddr *name, socklen_t *name_len);
 
+/**
+    \brief  Get the name of the connected peer socket.
+
+    This function retrieves the address of the peer connected to the socket
+    specified by the socket descriptor. The address is returned in the buffer
+    pointed to by the addr parameter, and the actual length of the address is
+    returned in the addrlen parameter.
+
+    \param  socket          A socket that is already connected to a peer.
+    \param  name            A pointer to a sockaddr structure where the peer 
+                            address will be stored.
+    \param  address_len     A pointer to a socklen_t variable that specifies the 
+                            length of the address structure.
+                            On return, it will contain the actual size of the 
+                            address returned.
+
+    \retval 0               On success.
+    \retval -1              On error, sets errno as appropriate, such as:
+                            - EBADF: The socket argument is not a valid file 
+                                     descriptor.
+                            - EFAULT: The addr argument points to memory not in a 
+                                      valid part of the process address space.
+                            - ENOBUFS: Insufficient resources were available in the 
+                                       system to perform the operation.
+                            - ENOTCONN: The socket is not connected.
+                            - ENOTSOCK: The socket argument does not refer to a 
+                                        socket.
+*/
+int getpeername(int socket, struct sockaddr *restrict name, 
+                socklen_t *restrict name_len);
```

Comment:
```suggestion
                socklen_t *__RESTRICT name_len);
```
Need to ALWAYS make sure to use the `#define` alias for it when within a public header file in KOS, because it's not actually supported by cuck-ass C++, which is going to shit itself when it encounters it.

Reply:
Thanks!